qiita_db/analysis.py

```diff
@@ -812,7 +812,7 @@ def build_files(self,
             # doing it here cause it's computational cheaper
             all_ids = list(chain.from_iterable([
                 samps for _, samps in viewitems(samples)]))
-            rename_dup_samples = ((len(all_ids) != len(set(all_ids))) or
+            rename_dup_samples = ((len(all_ids) != len(set(all_ids))) and
```

Comment:
The or made sense when you had only one kind of data. The and makes sense when you have multiple data. Once we allow for multiple references this is gonna change to be tested per data_type/reference. Thus, no more tests are needed. Please see my 2nd point of my initial comment.